aspnetcore-webapi/RateLimitingDotNET8/RateLimitingDotNET8/Program.cs

```diff
@@ -0,0 +1,156 @@
+        options.QueueProcessingOrder = QueueProcessingOrder.OldestFirst;
+        options.QueueLimit = slidingOptions.QueueLimit;
+    }));
+}
+
+static void TokenBucketRateLimiter(WebApplicationBuilder builder)
+{
+    var myOptions = builder.Configuration.Get<TokenBucketOptions>();
+
+builder.Services.AddRateLimiter(options => options
+    .AddTokenBucketLimiter(policyName: Policies.Token, options =>
+    {
+        options.TokenLimit = myOptions!.TokenLimit;
+        options.QueueProcessingOrder = QueueProcessingOrder.OldestFirst;
+        options.QueueLimit = myOptions.QueueLimit;
+        options.ReplenishmentPeriod = TimeSpan.FromSeconds(myOptions.ReplenishmentPeriod);
+        options.TokensPerPeriod = myOptions.TokensPerPeriod;
+        options.AutoReplenishment = myOptions.AutoReplenishment;
+    }));
+}
+
+static void ConcurrencyRateLimiter(WebApplicationBuilder builder)
+{
+    var myOptions = builder.Configuration.Get<MyRateLimiterOptions>();
+
+builder.Services.AddRateLimiter(_ => _
+    .AddConcurrencyLimiter(policyName: Policies.Concurrency, options =>
+    {
+        options.PermitLimit = myOptions!.PermitLimit;
+        options.QueueProcessingOrder = QueueProcessingOrder.OldestFirst;
+        options.QueueLimit = myOptions.QueueLimit;
+    }));
+}
+
+static void AuthorizationRateLimiter(WebApplicationBuilder builder)
+{
+    var authorizedLimiterOptions = builder.Configuration.Get<MyRateLimiterOptions>();
+
+    var unauthorizedLimiterOptions = builder.Configuration.Get<MyRateLimiterOptions>();
+
+    builder.Services.AddRateLimiter(limiterOptions =>
+    {
+        limiterOptions.RejectionStatusCode = StatusCodes.Status429TooManyRequests;
+        limiterOptions.AddPolicy(policyName: "jwt", partitioner: httpContext =>
+        {
+            var accessToken = httpContext.GetTokenAsync("access_token").Result
+                              ?? string.Empty;
+    
+            if (!StringValues.IsNullOrEmpty(accessToken))
+            {
+                return RateLimitPartition.GetFixedWindowLimiter(accessToken, options =>
+                    new FixedWindowRateLimiterOptions
+                    {
+                        QueueLimit = authorizedLimiterOptions!.QueueLimit,
+                        PermitLimit = authorizedLimiterOptions.PermitLimit,
+                        Window = TimeSpan.FromMinutes(authorizedLimiterOptions.Window),
+                    });
+            }
+    
+            return RateLimitPartition.GetFixedWindowLimiter("Anon", _ =>
+                new FixedWindowRateLimiterOptions
+                {
+                    QueueLimit = unauthorizedLimiterOptions!.QueueLimit,
+                    PermitLimit = unauthorizedLimiterOptions.PermitLimit,
+                    Window = TimeSpan.FromMinutes(unauthorizedLimiterOptions.Window),
+                });
+        });
+    });
+}
+
+static void ChainedRateLimiter(WebApplicationBuilder builder)
+{
```

Comment:
Let's follow consistency, do not hardcode the settings. Added required configuration in the appsettings file, then read and use it from there.